src/test/java/org/openrewrite/kotlin/KotlinTypeMappingTest.java

```diff
@@ -510,6 +512,98 @@ public K.When visitWhen(K.When when, AtomicBoolean found) {
             );
         }
 
+        @ParameterizedTest
+        @CsvSource({
+          "n++,kotlin.Int",
+          "--n,kotlin.Int",
+          "n += a,kotlin.Int",
+          "n = a + b,kotlin.Int{name=plus,return=kotlin.Int,parameters=[kotlin.Int]}"
```

Comment:
The J.Binary has a method type instead of a boolean. This should probably be detected as a non-overloaded method and assign the return type instead.